File: Lecture 11/Python Code/Tomato/models/Restaurant.py
class Restaurant:
    next_restaurant_id = 0

    def __init__(self, name, location):
        self.restaurant_id = Restaurant.next_restaurant_id + 1
        Restaurant.next_restaurant_id += 1
        self.name = name
        self.location = location
        self.menu = []

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_location(self):
        return self.location

    def set_location(self, location):
        self.location = location

    def add_menu_item(self, item):
        self.menu.append(item)

    def get_menu(self):
        return self.menu

File: Lecture 11/Java Code/Tomato/models/Restaurant.js
export class Restaurant {
    static nextRestaurantId = 0;

    constructor(name, location) {
        this.name = name;
        this.location = location;
        this.restaurantId = ++Restaurant.nextRestaurantId;
        this.menu = [];
    }

    getName() {
        return this.name;
    }

    setName(n) {
        this.name = n;
    }

    getLocation() {
        return this.location;
    }

    setLocation(loc) {
        this.location = loc;
    }

    addMenuItem(item) {
        this.menu.push(item);
    }

    getMenu() {
        return this.menu;
    }
}
